src/charm.py

```diff
@@ -1,104 +1,397 @@
+            return
+        if not self.model.get_relation("alertmanager"):
+            self.unit.status = BlockedStatus("Waiting for alertmanager relation to be created")
+            event.defer()
+            return
+        if not self._alertmanager_config_file_exists:
+            self.unit.status = WaitingStatus(
+                "Waiting for Alertmanager config file to be available"
+            )
+            event.defer()
+            return
+        config = remote_config_write.load_config_file(self.ALERTMANAGER_CONFIG_FILE)
+        templates = self._get_templates(config)
+        self._update_alertmanager_relation_data_bag_with_new_alertmanager_config(config, templates)
+
+    def _get_templates(self, config: dict) -> list:
+        """Prepares templates data to be put in a relation data bag.
+        If the main config file contains templates section, content of the files specified in this
+        section will be concatenated. At the same time, templates section will be removed from
+        the main config, as alertmanager-k8s-operator charm doesn't tolerate it.
+        It is also possible to configure templates through the charm's templates_file config
+        option.
+        In case templates are specified in both the main config file and the charm's config, config
+        file will take precedence.
+
+        Args:
+            config: Alertmanager config
+
+        Returns:
+            list: List of templates
+        """
+        templates = []
+        if config.get("templates", []):
+            for file in config.pop("templates"):
+                try:
+                    templates.append(remote_config_write.load_templates_file(file))
+                except FileNotFoundError:
+                    continue
+        elif self.config["templates_file"]:
+            templates.append(self.config["templates_file"])
+        return templates
+
+    def _update_alertmanager_relation_data_bag_with_new_alertmanager_config(
+        self,
+        config: dict,
+        templates: list,
+    ) -> None:
+        """Updates Alertmanager config and templates inside the relation data bag.
+
+        Args:
+            config: Alertmanager config
+            templates: Alertmanager templates
+
+        Returns:
+            None
+        """
+        alertmanager_relation = self.model.get_relation("alertmanager")
+        alertmanager_relation.data[self.app]["alertmanager_config"] = json.dumps(config)  # type: ignore[union-attr]  # noqa: E501
+        alertmanager_relation.data[self.app]["alertmanager_templates"] = json.dumps(templates)  # type: ignore[union-attr]  # noqa: E501
+
+    def _on_alertmanager_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Handles actions taken when Alertmanager relation joins.
+
+        Returns:
+            None
+        """
+        self._get_default_alertmanager_config(event)
+
+    def _get_default_alertmanager_config(self, event: RelationJoinedEvent) -> None:
```

Comment:
I don't thing `get` is the correct word, `push_default_config_to_workload` would make more sense

Reply:
Changed.